app/jobs/history/ledger_importer_job.rb

```diff
@@ -377,27 +374,73 @@ def import_history_effects_for_operation(sctx, scop, scresult, hop)
         })
       end
 
-      # unless scop.thresholds.nil?
-      #   effects.create!("account_thresholds_updated", source_account, {
-      #     # TODO: fill in details
-      #   })
-      #
-      #   # TODO: import signer_* effects for master key
-      # end
-
-      unless scop.set_flags.nil? && scop.clear_flags.nil?
-        effects.create!("account_flags_updated", source_account, {
-          # TODO: fill in details
+      thresholds_changed = scop.low_threshold.present? ||
+                           scop.med_threshold.present? ||
+                           scop.high_threshold.present?
+
+
+      if thresholds_changed
+        details = {}
+        details["low_threshold"]  = scop.low_threshold  if scop.low_threshold.present?
+        details["med_threshold"]  = scop.med_threshold  if scop.med_threshold.present?
+        details["high_threshold"] = scop.high_threshold if scop.high_threshold.present?
+        effects.create!("account_thresholds_updated", source_account, details)
+      end
+
+      flag_changes = {}
+      Stellar::AccountFlags.parse_mask(scop.set_flags || 0).each do |af|
+        flag_changes [af.name] = true
+      end
+      Stellar::AccountFlags.parse_mask(scop.clear_flags || 0).each do |af|
+        flag_changes [af.name] = false
+      end
+
+      if flag_changes.any?
+        effects.create!("account_flags_updated", source_account, flag_changes)
+      end
+
+      if scop.master_weight.present?
+        #TODO: BLOCKED stellar-core: differentiate signer_updated and signer_added
+        #for master signer
+        effect = scop.master_weight == 0 ? "signer_removed" : "signer_updated"
+
+        effects.create!(effect, source_account, {
+          public_key: Stellar::Convert.pk_to_address(source_account),
```

Comment:
Yeah, it definitely does.  However, that's actually a change to https://github.com/stellar/ruby-stellar-base, a dependency of horizon.

`Stellar::Convert.pk_to_address` is the current canonical helper for this functionality, so I'll keep it the same for now, but when I go about adding `address` to `Stellar::AccountID` I'll switch it over.

Thanks!